Practice/03_12/script.js

```diff
@@ -6,3 +6,15 @@
  * - Create several objects using the class.
  * - Test the objecs by calling their properties and using their methods in the console.
  */
```

Comment:
For this example, I'll use a `Car` class that has properties like `make`, `model`, `year`, and methods like `startEngine` and `stopEngine`.

### Car Class Definition

```javascript
class Car {
  constructor(make, model, year) {
    this.make = make;
    this.model = model;
    this.year = year;
    this.isEngineOn = false;
  }

  startEngine() {
    if (!this.isEngineOn) {
      this.isEngineOn = true;
      console.log(`${this.make} ${this.model} engine started.`);
    } else {
      console.log(`${this.make} ${this.model} engine is already on.`);
    }
  }

  stopEngine() {
    if (this.isEngineOn) {
      this.isEngineOn = false;
      console.log(`${this.make} ${this.model} engine stopped.`);
    } else {
      console.log(`${this.make} ${this.model} engine is already off.`);
    }
  }
}
```

### Creating Objects and Testing

Now, let's create several objects using the `Car` class and test them by accessing their properties and using their methods.

```javascript
// Creating car objects
const car1 = new Car("Toyota", "Corolla", 2020);
const car2 = new Car("Honda", "Civic", 2019);
const car3 = new Car("Ford", "Mustang", 2021);

// Accessing properties
console.log(`${car1.make} ${car1.model} (${car1.year})`);
console.log(`${car2.make} ${car2.model} (${car2.year})`);
console.log(`${car3.make} ${car3.model} (${car3.year})`);

// Testing methods
car1.startEngine();
car1.stopEngine();

car2.startEngine();
car2.startEngine(); // Trying to start the engine again
car2.stopEngine();

car3.startEngine();
car3.stopEngine();
car3.stopEngine(); // Trying to stop the engine again

// Output for each method call is printed to the console
```

### Explanation:

1. **Class Definition**:
   - The `Car` class has a constructor that initializes the `make`, `model`, `year`, and `isEngineOn` properties.
   - The `startEngine` method checks if the engine is off and starts it if true, otherwise, it indicates that the engine is already on.
   - The `stopEngine` method checks if the engine is on and stops it if true, otherwise, it indicates that the engine is already off.

2. **Creating Objects**:
   - We create three instances (`car1`, `car2`, `car3`) of the `Car` class, each representing a different car.

3. **Accessing Properties**:
   - We print out the `make`, `model`, and `year` properties of each car object to the console.

4. **Testing Methods**:
   - We test the `startEngine` and `stopEngine` methods on each car object. The console output will show the engine status and handle cases where the engine is already in the desired state.

This demonstrates the basics of object-oriented programming in JavaScript using a class to create and manage objects, as well as how to work with their properties and methods.